odoo/models.py

```diff
@@ -3144,6 +3145,15 @@ def unlink(self):
                 if attachments:
                     attachments.unlink()
 
+                if any(field.translate for field in self._fields.values()):
+                    # For the same reason, remove the relevant records in ir_translation
+                    translations = Translation.search([
+                        ('type', 'in', ['model', 'model_terms']),
+                        ('name', '=like', self._name+',%'),
+                        ('res_id', 'in', sub_ids)])
+                    if translations:
+                        translations.unlink()
```

Comment:
`unlink` works on an empty recordset iirc

Reply:
indeed but while it should not cause any issue on properly implemented unlink overrides, I wanted to avoid to trigger things like this ;-)
https://github.com/odoo/odoo/blob/7e9a13a64d71afc7d46d616824daf86938c9ee1a/addons/calendar/models/calendar.py#L523